### ISSUE TITLE
Kwargs after ParamSpec.args should be tolerated

### ISSUE THREAD

**Issue by kdkavanagh** [python/mypy#19506]
**Bug Report**
I believe mypy's handling of `ParamSpecs` is slightly too restrictive.  Currently, it validates that there are no args _or kwargs_ after a `ParamSpec.args` argument other than `**kwargs`.  I believe it's tolerable to have _kwargs_ follow a `ParamSpec.args`

Related to #18278

**To Reproduce**

```python
def f5(f: Callable[P, int], *args: P.args, extra_regular_arg: int, **kwargs: P.kwargs) -> int: ...  # E: Arguments not allowed after ParamSpec.args

def f6(f: Callable[P, int], *args: P.args, extra_kwarg: int = 123, **kwargs: P.kwargs) -> int: ...  # E: Arguments not allowed after ParamSpec.args
```

**Expected Behavior**
The first function above should continue to fail, while the second function should not throw an error

**Comment by brianschubert** [comment]
Thanks for the issue! However, the behavior you suggest is not permitted by the [typing spec](https://typing.python.org/en/latest/spec/generics.html#paramspec#:~:text=Placing%20keyword-only,forbidden), which states:
> Placing keyword-only parameters between the `*args` and `**kwargs` is forbidden.

Both `extra_regular_arg` in `f5` and `extra_kwarg` in `f6` are keyword-only parameters, so mypy is correct to reject them. The reason for this restriction explained in the Rejected Alternatives section of the original PEP: https://peps.python.org/pep-0612/#concatenating-keyword-parameters
